cmd/ttn-lw-cli/commands/end_device_templates.go

```diff
@@ -0,0 +1,436 @@
+				return err
+			}
+			var startDevEUI types.EUI64
+			if err := startDevEUI.UnmarshalText([]byte(startDevEUIHex)); err != nil {
+				return err
+			}
+			devEUIInt := binary.BigEndian.Uint64(startDevEUI[:])
+
+			count, _ := cmd.Flags().GetInt("count")
+			for {
+				var template ttnpb.EndDeviceTemplate
+				if _, err := inputDecoder.Decode(&template); err != nil {
+					if err == stdio.EOF {
+						return nil
+					}
+					return err
+				}
+
+				for i := 0; i < count; i++ {
+					res := template
+
+					var devEUI types.EUI64
+					binary.BigEndian.PutUint64(devEUI[:], devEUIInt)
+					devEUIInt++
+
+					res.EndDevice.DeviceID = fmt.Sprintf("eui-%s", strings.ToLower(devEUI.String()))
+					res.EndDevice.JoinEUI = &joinEUI
+					res.EndDevice.DevEUI = &devEUI
+					res.FieldMask.Paths = ttnpb.BottomLevelFields(append(res.FieldMask.Paths,
+						"ids.device_id",
+						"ids.join_eui",
+						"ids.dev_eui",
+					))
+
+					if err := io.Write(os.Stdout, config.OutputFormat, &res); err != nil {
+						return err
+					}
+				}
+			}
+		},
+	}
+	endDeviceTemplatesListFormats = &cobra.Command{
+		Use:               "list-formats",
+		Short:             "List available end device template formats",
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dtc, err := api.Dial(ctx, config.DeviceTemplateConverterGRPCAddress)
+			if err != nil {
+				return err
+			}
+			res, err := ttnpb.NewEndDeviceTemplateConverterClient(dtc).ListFormats(ctx, ttnpb.Empty)
+			if err != nil {
+				return err
+			}
+			return io.Write(os.Stdout, config.OutputFormat, res)
+		},
+	}
+	endDeviceTemplatesConvertCommand = &cobra.Command{
+		Use:               "convert [format-id]",
+		Short:             "Convert data to an end device template",
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			formatID := getTemplateFormatID(cmd.Flags(), args)
+			if formatID == "" {
+				return errNoTemplateFormatID
+			}
+			data, err := getDataBytes("", cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			dtc, err := api.Dial(ctx, config.DeviceTemplateConverterGRPCAddress)
+			if err != nil {
+				return err
+			}
+			stream, err := ttnpb.NewEndDeviceTemplateConverterClient(dtc).Convert(ctx, &ttnpb.ConvertEndDeviceTemplateRequest{
+				FormatID: formatID,
+				Data:     data,
+			})
+			if err != nil {
+				return err
+			}
+
+			for {
+				dev, err := stream.Recv()
+				if err == stdio.EOF {
+					return nil
+				}
+				if err != nil {
+					return err
+				}
+				if err := io.Write(os.Stdout, config.OutputFormat, dev); err != nil {
+					return err
+				}
+			}
+		},
+	}
+	endDeviceTemplatesMapCommand = &cobra.Command{
+		Use:   "map [flags]",
+		Short: "Map end device templates",
+		Long: `Map end device templates
+
+This command matches the input template with the mapping file to create new
+templates. The mapping file contains end device templates in JSON format.
```

Comment:
I would use "the same format as input". The inputDecoder will also be able to decode from yaml/toml/...